src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -650,7 +657,7 @@ private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, Dat
         internal ImageClassificationTransformer(IHostEnvironment env, Session session, string[] outputColumnNames,
             string[] inputColumnNames, string modelLocation,
             bool? addBatchDimensionInput, int batchSize, string labelColumnName, string finalModelPrefix, Architecture arch,
-            string scoreColumnName, string predictedLabelColumnName, float learningRate, DataViewSchema inputSchema, int? classCount = null, bool loadModel = false,
+            string scoreColumnName, string predictedLabelColumnName, float learningRate, DataViewSchema inputSchema, int? classCount = null, string[] keyValueAnnotations = null, bool loadModel = false,
```

Comment:
>keyValueAnnotations [](start = 150, length = 19)

The transformer can be created in one of two ways: Either by calling `Fit` on the estimator, or by deserializing from file. In the second case, the key values come from reading them from the model as you have written. In the first case, you need to get them from `inputSchema` by getting the key value annotations of the training label column.

Reply:
Notice that it is possible that the label column does not contain key values annotations, which means that the PredictedLabel column will not have them either. You have to account for that at serialization/deserialization time (for example, by serialiazing the number of key value strings, and only then serializing the strings them selves).

---
In reply to: [327481050](https://github.com/dotnet/machinelearning/pull/4228#discussion_r327481050) [](ancestors = 327481050)